Lab 3: Genotype Imputation and Haplotype Phasing with IMPUTE2
========================================================
*Created by Max Winston; modified by Charles Washington III and Ankeeta Shah*

Genotype imputation relies on statistical inference and known haplotypes in a population to estimate unobserved genotypes. Often, particularly in epidemiology and quantitative genetics, a researcher may want to identify a specific location in the genome where genetic variation is associated with a particular phenotype, but they may only have access to sparse genotype data (e.g. data from a SNP genotyping array as opposed to complete sequence/whole genome sequencing data). At the same time, dense genotype data may exist for a reference panel of individuals (e.g. from the HapMap project or 1000 Genomes for humans).  By leveraging the information in the reference panel to impute gentoypes into the inference panel, a reseracher can interrogate more variants for association with the trait.  In some cases, this reveals associations that would not be detected directly in the sparse data.  In other cases, it may help refine the location of a causal variant. 

The imputation process relies on haplotype phasing, which is the statistical estimation of haplotypes from genotype data. Both of these steps can be accomplished from within IMPUTE2, and will be the focus of today's lab. By the end of the lab you should be able to:

- **Understand basic workflow and syntax of IMPUTE2**  
- **Execute basic imputation in IMPUTE2**
- **Navigate basic commands for file conversion using GTOOL**  
- **Understand key assumptions of imputation**
- **Utilize IMPUTE2 and PLINK to search for putative associations in imputed SNPs**

Section 1: Basics of IMPUTE2
-------------------------------

### *1.1: BASH commands useful for this lab*  

Command     |     Description
----------  |   ----------------------------------
**cut**     |   Extract sections from each line of input stream (i.e. file).  
**sed**     |   Stream editor for performing basic text transformations on an input stream.
**paste**   |   Merges lines of files together.
**wc**      |   Word, line, character, and byte count.  
**mkdir**   |   Make directory.
**head**    |   Output first part of files.
**grep**    |   Prints lines matching a pattern (helpful for manipulating large files).

### *1.2: Standard IMPUTE2 arguments*

Argument      |  Input Type     |       Description
-----------   | -------         |     -----------------------------------
**-g**        |   .gens         |   File containing study genotypes needing imputation or phasing.  
**-m**        |   .map          |   Fine-scale recombination map for the region to be analyzed.
**-int**      |   [*interval*] |  Genomic interval to use for imputation inference.
**-h**        |   .haps         |  File of known haplotypes, with one row per SNP and one column per haplotype (alleles must be coded 0 or 1)
**-l**        |  .legend        |   Legend file(s) with information about the SNPs in the haplotypes file. 
**-strand_g**  |   .strand       |   File containing strand orientation of the inference set. 
**-Ne**        |   [*pop size*]  |   Parameter controlling effective population size. 


### *1.3: Downloading and Running IMPUTE2*

First, in case you don't have IMPUTE2 downloaded from the internet from Lab 1 or you can't locate it, you can **follow the command below to download and unpack it**: 

```{r, eval=FALSE}

curl https://mathgen.stats.ox.ac.uk/impute/impute_v2.3.2_x86_64_static.tgz > impute2.tgz              
tar -zxvf impute2.tgz  

```


### *1.4: File formatting for IMPUTE2*  

There are several file formats necessary for IMPUTE2, which are listed above in Section 1.2. It is helpful to understand the basic formatting for these input files and their importance in the imputation, as this well facilitate manipulating them in the future. 

#### *1.4.1: GENS file*  

The GENS file is the general file type for genotype data in IMPUTE2. Each row represents a different SNP, and the first five columns are: (1) **SNP ID**, (2) **rsID**, (3) **Position (bp)**, (4) **Major Allele**, (5) **Minor Allele**. The sixth column onward represents the genotypic information for all of the individuals, where each individual has three columns representing the probabilities of each genotype. This file does NOT have a header row.

#### *1.4.2: MAP file*  

The MAP file provides a fine-scale recombination map with three columns: (1) *physical position (in base pairs)*, (2) *recombination rate between current position and next position in map (in cM/Mb)*, (3) *and genetic map position (in cM)*. This information is critical in the process of imputation in that altering rates of recombination in a region will significantly change the likelihoods for imputed SNPs. For example, if there are high rates of recombination between SNP A from your SNP array and desired imputed SNP B, you may have less statistical power or certainty to impute that SNP depending on the depth of your reference haplotypes. Note that this file DOES have a header row. You can always check if a file has one (and you should need to a few times in this lab) simply by opening it with your text editor of choice.

*Problem 1*  
All the files that begin with the label "example.chr22" are from a particular stretch of DNA on chromosome. What position in this stretch has the highest recombination rate? What is it (please include units)?

#### *1.4.3: HAPS file*  

The HAPS file is a file of reference haplotypes (usually binary), often from a larger project (i.e. 1000 Genomes), which is used during imputation to compare against haplotypes from the inference panel to estimate probabilities for imputed SNPs. Each line represents a distinct haplotype to be used as a reference.  

*Problem 2*  
How many haplotypes are provided in the example file *example.chr22.hm3.haps*? How many in the example file *example.chr22.1kG.haps*?

#### *1.4.4: LEGEND file*  

The LEGEND file provides information on the HAPS file, so there must be a one-to-one correspondence. Thus, it is generally not wise to edit the HAPS or the LEGEND file, but if you were to do this you would need to edit them in concert. Each LEGEND file provides four columns of information, with each row corresponding to the same row in the HAPS file. The four columns of information are: (1) **rs ID**, (2) **Position (bp)**, (3) **Major Allele**, (4) **Minor Allele**. 

*Problem 3*  
What is the major allele for SNP "rs1669115"? What is the minor allele?

#### *1.4.5: STRAND file*  

The STRAND file is a pretty simple but critical file that ensures the reference set and the inference set are comparing the same strand of DNA (similar 5' to 3' distinction). This avoids confusion about ambigous genotype calls -- for example, what one investigator calls a GG call could be called a CC by another investigator looking at the opposing strand of DNA. To denote which strand we use relative to the human reference sequence, a "+" denotes the same strand as the reference genome, and a "-" denotes the opposing strand. There are only two columns: (1) **Position (bp)**, (2) **Strand**. You really shouldn't be altering this file unless completely necessary, as it can introduce some very big errors. Although this won't be necessary for today's lab, it is essential to have the strand matched between the reference and imputation panel when imputing data and one should be on guard for "strand flips" as an explanation for anomolous results.

### *1.5: Running the imputation*  

For getting the hang of running IMPUTE2, we will be imputing the area on the 22nd chromosome between the SNPs sequenced in the example files *"example.chr.study"*. These files specify the necessary information for our **inference panel**. Likewise, we have a **reference panel** in this same region specified in the files starting with *"example.chr.1kg"*. **Save the following command into a new file named "execute_impute.sh"**: 

```{r, eval=FALSE}
./impute2 -m ./Example/example.chr22.map -h ./Example/example.chr22.1kG.haps -l ./Example/example.chr22.1kG.legend -g ./Example/example.chr22.study.gens -strand_g ./Example/example.chr22.study.strand -int 20.3e6 2\
0.6e6 -Ne 20000 -o ./Example/example.chr22.one.phased.impute2

```

Note that you will need to be in the directory where you have unpacked the impute2 tarball, and where you can see the *Example* directory listed out, in order for this call to work. Now, after saving this command--which should all be on a single line (don't press enter)--into a text file "execute_impute.sh", **execute the command by typing**:

```{r, eval=FALSE}

bash execute_impute.sh

```

This should create five files in the specified directory (*Example*), all of which should start with *"example.chr22.one.phased.impute2"*. The file with the actual imputed genotypes in GENS format should be the file *just* called *"example.chr22.one.phased.impute2"*. Other supplementary output files have underscores and then IDs: **"warning"** gives any errors during the imputation process, **"summary"** is just a log file, **"info"** gives a variety of information for each SNP as demonstrated in the header, and **"info by sample"** gives information by individuals instead of by SNP. 

*Problem 4*  
Open the summary file in a text editor and scroll down to line 60 (L60), where it should mention 'flipping strands'. What does it say and how would you interpret this? 

### *1.6: Using GTOOL post-imputation*

GTOOL is a command-line program that facilitates conversion between PLINK and IMPUTE2 file formats. The program is well documented, and it is recommended that you visit the documentation on the website for any purposes outside this lab (http://www.well.ox.ac.uk/~cfreeman/software/gwas/gtool.html).

#### *1.6.1: Downloading GTOOL*  

Downloading and unpacking GTOOL is as easy as any other *tarball* download, and can be **performed with the following command**:

```{r, eval=FALSE}

curl https://www.well.ox.ac.uk/~cfreeman/software/gwas/gtool_v0.7.5_x86_64.tgz > gtool.tgz              
tar -zxvf gtool.tgz  

```

These commands should create both an executable file "gtool" as well as an "example/" directory. **Move these into a new directory you make named "GTOOL"**. 

#### *1.6.2: Basic GTOOL usage*  

Here we will practice using GTOOL on one of the small GENS files with only 33 markers from the IMPUTE2 *"Example/"* directory, *"example.chr22.study.gens"*. We will need to provide this file, as well as a list of samples "example.study.samples" in the same directory. **Save the following command into a text file and name the file "gtool_execute.sh"**:


```{r, eval=FALSE}

/home/washingtonc/Lab_3/impute_v2.3.2_x86_64_static/GTOOL/gtool -G --g ../Example/example.chr22.study.gens --s ../Example/example.study.samples --ped ../Example/real_small.ped --map ../Example/real_small.map --threshold 0.9

```

Don't forget that you will need to modify this command to find your own gtool executable, which should be in one of your directories, not mine (/home/washingtonc/...). You'll also definitely need to modify the paths to the example.chr22 and example.study.samples files, based on where those are in your workspace. Also note that the *real_small* files are both being created with this call--so don't be confused looking for them.

For the rest of the lab, make sure to **save all of your GTOOL commands in the file you just created (gtool_execute.sh)**. Before running your GTOOL command, take a minute to look at the arguments of gtool. One important parameter is the **"--threshold"** argument. This tells GTOOL the likelihood cut-off for a genotype at any given SNP. For example, in our command, any likelihood below 0.9 at a given SNP will produce a "missing genotype" for PLINK. **Run the "gtool_execute.sh" script with the following command**:

```{r, eval=FALSE}

bash gtool_execute.sh

```

*Problem 5*  
Recalling your knowledge of the structure of PED files, what do you notice about both the phenotype and gender information in the file we just converted ("real_small.ped")?

Section 2: Simulating a Case-Control Scenario with Imputation
-------------------------------

In this section we will use the skills we have learned in the first section to simulate a scenario in which you have localized a region through previous work on the 22nd chromosome that is associated with Disease XXX. You will need to use commands from PLINK, GTOOL, IMPUTE2, and BASH to work through this scenario (consult previous labs if you have trouble remembering how to run certain code), in which you will run an association, identify the SNP on your array, convert data to IMPUTE2 format, and impute unobserved SNPs for another association in an attempt to get a more precise estimate of the associated SNP.

### *2.1: Running an initial GWAS*

Here, you will run an initial GWAS on a set of 33 markers in the file *"real_small_final"*. For the sake of this exercise, assume that these 33 markers restricted to a small section of chromosome 22 (20303319-20596808bp) are part of a larger GWAS that you ran with a total of 100,000 SNPs. In that study, you identified this region, and now you want to pursue your study further with imputation. Just to check that everything is working correctly, you want to check the reduced set (33 markers) with a GWAS.

In order to reduce the work for this lab, I have already provided you *"real_small_final"* files (on Canvas) that have been converted from the results of GTOOL in Section 1.6.2 to add our disease phenotype. Although I have done this for you, I would like you to understand how to do this so you can do it yourself. **Open the script "PED_change_small.sh" (also on Canvas) in a text editor**. 

*Problem 6*  
In a step-by-step fashion, describe what steps 1, 2, 3, & 5 in **"PED_change_small.sh"** are doing in order to create *"real_small_final.ped"*.

*Problem 7*  
Run a basic association (--assoc) on the *"real_small_final"* dataset I provided you. First, you'll need to create a BED file and then run the association. Save your commands in a BASH script named **"GWAS_execute.sh"** and provide them. Hint: You may need the "--allow-no-sex" flag. 

*Problem 8*  
According to this analysis, what is the most highly associated SNP? What is the unadjusted p-value? Considering what you know about how we found this SNP (see above), what do you think this SNP's adjusted (Bonferroni-correction) p-value should be? 

### *2.2: Manhattan plot of initial GWAS*

*Problem 9*  
Use the knowledge gained in Lab 2 to export the *.assoc* file and produce a Manhattan plot in RStudio. Make sure to use the appropriate Bonferroni-correction (think about how you identified this region as well) for the Manhattan plot.

### *2.3: Imputation and conversion*

Imputation for this set of 33 markers was completed in Section 1.5. The file you produced should be called **"example.chr22.one.phased.impute2"**. Let's take this GENS file and convert it to PED format. 

*Problem 10*  
Using the first command in *"gtool_execute.sh"* as a guide, write and provide a second command in the script that will convert this file **"example.chr22.one.phased.impute2"** to PED format, and name this output *"test_out"*. You can "comment out" (turn off) the first command by putting a # in front of it. Do this and then run your bash script.

Finally, we need to modify this PED file to insert our phenotype into it.

*Problem 11*  
Use **"PED_change_small.sh"** as a guide to create and provide your own BASH script named **"PED_change.sh"** that modifies your *"test_out"* PED file from GTOOL by inserting *PHENOTYPE.txt* into it, naming it *"real_test.ped"*. (Hint: You don't need to change Step 4 at all, and the **PHENOTYPE.txt** file is also on Canvas). Make sure you rename your MAP file to match your new PED file.

### *2.4: Running a GWAS with Imputed SNPs*

Once you have your converted and reformatted PED and MAP files named *"real_test"*, we can now make our BED and FAM files which will be used for a GWAS. 

*Problem 12*  
Make a BED and FAM file from the PED and MAP files named *"real_test"*, then run a basic association on these new files. Save these commands in your **"GWAS_execute.sh"** script and provide them. 

*Problem 13*  
Use your BASH knowledge to find the most highly associated SNP in the *"real_test.assoc"* file. What SNP is it? What is the non-adjusted p-value? (Hint: using the -g flag with the *"sort"* command will deal with some issues that a basic numerical sorting with -n tends to have with p-values that have exponents attached to them. There's also another way to deal with this issue while still utilizing the -n flag instead of -g, which is based on the p-values' relationship to another column within this file). 

### *2.5: Manhattan Plot of Imputed GWAS*

Lastly, we will visualize the results of our GWAS with imputed SNPs by exporting our *.assoc"* file to our computer and plot it using RStudio. 

*Problem 14*  
Similar to Problem 9, produce a Manhattan plot for the file in RStudio. Describe the differences between this plot and the plot from Problem 9 in a couple of sentences.

References
----------------------------
**B. N. Howie, P. Donnelly, and J. Marchini (2009).** A flexible and accurate genotype imputation method for the next generation of genome-wide association studies. *PLoS Genetics* 5(6): e1000529
