---
title: Final Exam
author: Ansel George
output: pdf_document
---

# Problem 1

Consider a scenario in which genotypes are observed for pairs of siblings who
are second cousins. Each pair of siblings will have long chromosomal segments
that are inherited from their shared great grandparent (tracts that are
identical by descent, IBD). Suppose we would like to design a Hidden Markov
Model that could be used to infer the location of IBD tracts.

a) Sketch the model, defining your hidden and observed random variables.

b) Which algorithm would you use to identify the IBD segments?

c) If two markers are separated by 0.05 cM, how could that used to inform
   transition probabilities in the model?

d) Extra-credit: Define the emission probabilities assuming known allele
   frequencies, and assuming Hardy-Weinberg proportions.


# Problem 2

Researchers often study isolated populations, e.g. Icelandic people. Typically,
a small founder population moved to a place, and then became isolated (limited
genetic flow) from the larger world. In this problem, we will explore the use
of isolated populations for studying human genetics. (Feel free to read
published work using this type of study design, e.g. papers from deCODE
Genetics.)

a) What are the main advantages of using isolated populations for studying
   genetics of human diseases?

b) Imputation is particularly effective for isolated populations. Typically
   researchers sequence a small number of individuals and then use that as a
   reference panel to impute more samples. Explain why it is easier to impute
   isolated populations.

c) Unlike standard GWAS, the individuals of a population are not unrelated. How
   would you perform association analysis of a SNP? (5 points)

d) Suppose we study the heritability of a complex trait. Do you expect
   heritability from isolated populations to be higher or lower or the same,
   comparing with studies using unrelated subjects?


# Problem 3

A researcher hypothesizes that longevity is influenced by genetics, and is
writing a proposal to perform a genome-wide association study of longevity. In
addition, s/he has access to a number of publicly available datasets such as
ENCODE, GTEx and results from model organism studies of longevity (e.g.
knockout of certain genes extends the lifespan of a model species). You will
help the research to develop the proposal, which should address the following
questions:

a) A simple definition of phenotype would be the life-span of an individual.
   But the challenge is that it is generally unknown until the person dies, at
   which point it is difficult to perform any studies. Propose a study design
   that overcomes this problem.

b) Unlike many other phenotypes, the primary tissue types related to longevity
   are not well-defined. Propose a strategy to address this question using the
   results from GWAS and additional resources.

c) How will you prioritize or follow up with the loci you find to be associated
   with longevity? You can propose any computational and experimental analysis,
   as long as it is practical.

d) In the case that the study is underpowered, you may not find many loci. What
   strategies can you use to boost the power?

e) One question about longevity is whether some people live very long because
   they have fewer disease variants or they have some extra protective
   variants. Propose some idea to compare the two hypotheses.


# Problem 4

In this problem, we explore the importance of considering the power when
comparing findings across multiple studies. As discussed in the class, cis-eQTL
of different tissues are often shared. Suppose in the first tissue, we found
$n$ eQTL above our significance threshold, $p < \lambda$. In the second tissue,
$k$ out of these $n$ eQTL are statistically significant at $p < \lambda$.

a) Can we say that the true proportion of cis-eQTL that are shared with the
   second tissue is $k/n$? If not, why?

b) Assume we know that the power in the eQTL study of the second tissue is
   $\beta$, at $p$- value threshold $\lambda$. Use these numbers to estimate
   how often a cis-eQTL in the first tissue is also an eQTL in the second
   tissue (Hint: some of the $n$ SNPs found in the first tissue are true eQTL
   in the second tissue, other are not. For each of these two sets of SNPs,
   what is the probability that a SNP has $p$-value below $\lambda$ in the
   second tissue?).


# Problem 5

Let's analyze real GWAS results and write a summary. Pick a phenotype from UK
Biobank here
(\url{https://docs.google.com/spreadsheets/d/1b3oGI2lUt57BcuHttWaZotQcI0-
mBRPyZihz87Ms_No/edit?usp=sharing}) and download the GWAS summary statistics.
Upload summary results to FUMA (\url{http://fuma.ctglab.nl/}; you need to
create an account). Write a report on your findings with the sections (one
paragraph each) listed below using this as a guide:
\url{http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1005619}.

a) Abstract

b) Introduction: Explain why you picked this phenotype and why it may be
   important.

c) Results: Include plots from the genome-wide plots tab from FUMA with your
   interpretation. Gene set analysis may sometimes provide you with interesting
   stories to tell.

d) Run S-PrediXcan (https://github.com/hakyimlab/MetaXcan) using your favorite tissue to
   find gene level associations and discuss the results.

e) Discussion
